sensu/plugins/check-swift-dispersion.py

```diff
@@ -35,22 +35,37 @@ def setup(self):
                                  help='Object warning dispersion percent')
 
     def run(self):
-
         output = None
-        try:
-            output = check_output(['swift-dispersion-report', '-j'],
-                                  stderr=STDOUT)
-
-            # strip any ERRORs that come through
-            p = re.compile(r'(\{.*\})')
-            m = p.search(output)
-            output = m.group(1)
-
-        except CalledProcessError as e:
-            self.critical("Unable to run swift-dispersion-check: %s %s" %
-                          (e, output))
-            return
-
+        # This should at most run twice when populate is not
+        # run. If populate is run error message would be different
+        # causing check to return immediately
+        while True:
+            proc_rep = Popen(['swift-dispersion-report', '-j'],
```

Comment:
Changed check_output to Popen to capture output, error. check_output raises an exception and error is not properly collected